libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtils.kt

```diff
@@ -1567,12 +1568,45 @@ internal object MapboxRouteLineUtils {
      * Returns a percentage offset of the fork intersection along the alternative route's geometry.
      */
     internal fun getAlternativeRouteDeviationOffsets(
-        metadata: AlternativeRouteMetadata
+        metadata: AlternativeRouteMetadata,
+        distancesProvider: (NavigationRoute) -> RouteLineGranularDistances? =
+            distinctGranularDistancesProvider
```

Comment:
This injection is a little bit forced, but I see not other quick way to be able to test this function efficiently due to the static nature of the whole `MapboxRouteLineUtils`.

We're starting to accumulate some debt in here and when time allows, we need to refactor our caching strategies to keep things maintainable. I'll follow up with a ticket.